File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaTenantModuleRepository.java
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories;

import java.util.List;
import java.util.UUID;

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.TenantModuleEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface JpaTenantModuleRepository extends JpaRepository<TenantModuleEntity, UUID> {
    List<TenantModuleEntity> findAllByTenantId(UUID tenantId);

    List<TenantModuleEntity> findAllByTenantIdAndEnabled(UUID tenantId, boolean enabled);

    TenantModuleEntity findByTenantIdAndModuleId(UUID tenantId, UUID moduleId);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaTenantModuleRepository.kt
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.TenantModuleEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface JpaTenantModuleRepository : JpaRepository<TenantModuleEntity, UUID> {
    fun findAllByTenantId(tenantId: UUID): List<TenantModuleEntity>
    fun findAllByTenantIdAndEnabled(tenantId: UUID, enabled: Boolean): List<TenantModuleEntity>
    fun findByTenantIdAndModuleId(tenantId: UUID, moduleId: UUID): TenantModuleEntity?
}
